qualification/Jenkinsfile

```diff
@@ -73,20 +73,33 @@ pipeline {
          script {
            sh 'pip install -r qualification/requirements.txt'
            sh 'pip install --no-deps ".[qualification]" && pip check'
-           try {
-             sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest'
-           }
-           finally {
-             COMMIT_ID = sh(script: 'qualification/report/generate_pdf.py report.json report.pdf -c', returnStdout: true)
-             sh(script: 'qualification/report/generate_pdf.py report.json procedure.pdf --generate-procedure-doc')
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf, report.json', reportTitles: 'report.pdf, report.json'])
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Procedure', reportDir: '', reportFiles: 'procedure.pdf', reportTitles: 'procedure.pdf'])
-             currentBuild.displayName = "#${BUILD_NUMBER} (${COMMIT_ID})"
-           }
+           sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --suppress-tests-failed-exit-code --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest --junitxml=result.xml'
          }
        }
-    }
-  }
+     }
+
+     stage('Publish results') {
+       steps {
+         script {
```

Comment:
Most of this block doesn't need to be wrapped in `script`. It's for when you need an escape hatch to write general Groovy code instead of just defining steps declaratively. I *think* the only bits that need to be wrapped in `script` are the two lines involving COMMIT_ID. See https://www.jenkins.io/doc/book/pipeline/syntax/#script.